### PR TITLE
[IMP] oca_publish_modules: tool for publishing modules

### PR DESCRIPTION
**Pull request by alexey-pelykh**
@pedrobaeza

### CODE REVIEW

File: tools/oca_publish_modules.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
```
Comment (elicoidal) [inline tools/oca_publish_modules.py]: no need for P3

File: tools/oca_publish_modules.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    login_button.click()
+    wait.until(lambda driver: driver.current_url == 'https://apps.odoo.com/apps')
+
+def register_repository(driver, repository):
+    driver.get('https://apps.odoo.com/apps/upload')
+    url_field = driver.find_element_by_id('url')
+    url_field.clear()
+    url.send_keys('https://github.com/OCA/' + repository)
+    submit_button = driver.find_element_by_id('apps_submit_repo_button')
+    submit_button.click()
```
Comment (elicoidal) [inline tools/oca_publish_modules.py]: What if there is an error? (eg: timeout, the repo is already submitted, etc.)

Comment (alexey-pelykh) [reply tools/oca_publish_modules.py]: This method is checked when there's need, all other issues - exception and crash

**Comment by pedrobaeza** [comment]
I have improved/fixed the flow according my research, but thanks for initial idea, @alexey-pelykh !

**Comment by pedrobaeza** [comment]
This is now completed and green.

**Comment by pedrobaeza** [comment]
Good news! Rémi Rahir from Odoo has told me several tricks for improving this through JSON endpoints, so I will try soon to implement these suggestions.

**Comment by alexey-pelykh** [comment]
@pedrobaeza at what stage are we on this one?

**Comment by pedrobaeza** [comment]
Stalled

**Comment by alexey-pelykh** [comment]
@pedrobaeza maybe let's just continue the work w/o those APIs?

**Comment by pedrobaeza** [comment]
Now it's in @simahawk's hands

**Comment by simahawk** [comment]
@alexey-pelykh @pedrobaeza is not in my hands: if this is something that is working, please, feel free to finish it.
We can use it and if needed we'll find a better solution in the future.
If you think you are too far away from something usable, that's another story :)
What I've done so far is just fixing manually many missing repos.

**Comment by alexey-pelykh** [comment]
@simahawk it was working, then there was a hope for APIs, but I'm not sure if that progressed anywhere. My vision is to rebase this, check that it's still working and do the magic of adopting it. But I need someone to help testing, @pedrobaeza helped me last time and contributed a lot to the original PR

**Comment by pedrobaeza** [comment]
Sorry, but I'm not able to do it. Too many hours last time and having our own apps store, I'm personally not making any attention to Odoo's one.

**Comment by alexey-pelykh** [comment]
@pedrobaeza that makes perfect sense, yet I don't really have the means to test it, so unless someone with means will assist (anyone, really), I won't be able to push this forward.

**Comment by sbidoul** [comment]
@alexey-pelykh I'm not a huge fan of the Odoo app store either but I think it's important OCA is present there too. if you rebase we'll find a way to test it. Either myself, @simahawk or @gurneyalex I guess.

**Comment by alexey-pelykh** [comment]
@sbidoul neither am I, but that's a free marketing :D sure, I'll rebase and let's make some tests

**Comment by simahawk** [comment]
same story here: if I land on odoo.com is because of clicks on search engine results or shared links :smile: 
Thanks guys

**Comment by simahawk** [comment]
@alexey-pelykh ciao :) Have you found time to make some tests?

**Comment by alexey-pelykh** [comment]
@simahawk nope, @sbidoul mentioned that OCA will find a way to test if I rebase, so I rebased :D and rebased again :D

**Comment by sbidoul** [comment]
@alexey-pelykh thanks a lot for this. I tweaked it a bit to make it work in the current state of the universe: https://github.com/brainbeanapps/maintainer-tools/pull/1

I published the current status at https://odoo-community.org/apps.odoo.com/

**Comment by sbidoul** [comment]
@alexey-pelykh you you mind giving OCA/maintainer-tools maintainers write access to this branch?

File: tools/publish_modules.py [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+    url_field.clear()
+    url_field.send_keys(repository)
+    submit_button = driver.find_element_by_id('apps_submit_repo_button')
+    submit_button.click()
+
+
+def scan_repository(driver, org, repository, branch, force_scan, scan_skip_empty):
+    wait = WebDriverWait(driver, 300)
+    for protocol in ('https', 'ssh'):
+        repository_url = url(repository, protocol='ssh', org_name=org) + '#' + branch
```
Comment (sbidoul) [inline tools/publish_modules.py:185]:
```suggestion
        repository_url = url(repository, protocol=protocol, org_name=org) + '#' + branch
```

**Comment by alexey-pelykh** [comment]
@sbidoul I would love to, but no such checkbox for me :(

**Comment by sbidoul** [comment]
Ok, I'll merge then, and we can iterate from here. Thanks a lot!

**Comment by sbidoul** [comment]
I created #462 and #463 as possible improvements in case anyone feels inspired.

**Pull request merged**